runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -2157,30 +2184,36 @@ void
 TR_RelocationRecordInlinedMethod::preparePrivateData(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget)
    {
    J9Method *ramMethod = NULL;
-   bool failValidation = true;
+   bool inlinedSiteIsValid = inlinedSiteValid(reloRuntime, reloTarget, ((TR_OpaqueMethodBlock **)&ramMethod));
+
+   if (reloRuntime->comp()->getOption(TR_UseSymbolValidationManager) && !ramMethod)
+      {
+      TR_ASSERT(false, "inlinedSiteValid should not return a NULL method when using the SVM!\n");
+      reloRuntime->comp()->failCompilation<J9::AOTSymbolValidationManagerFailure>("inlinedSiteValid returned NULL method");
+      }
 
-   if (validateClassesSame(reloRuntime, reloTarget, ((TR_OpaqueMethodBlock **)&ramMethod)) &&
-       !reloRuntime->options()->getOption(TR_DisableCHOpts))
+   if (ramMethod && !reloRuntime->options()->getOption(TR_DisableCHOpts))
       {
       // If validate passes, no patching needed since the fall-through path is the inlined code
       fixInlinedSiteInfo(reloRuntime, reloTarget, (TR_OpaqueMethodBlock *) ramMethod);
-      failValidation = false;
       }
 
    TR_RelocationRecordInlinedMethodPrivateData *reloPrivateData = &(privateData()->inlinedMethod);
    reloPrivateData->_ramMethod = (TR_OpaqueMethodBlock *) ramMethod;
-   reloPrivateData->_failValidation = failValidation;
-   RELO_LOG(reloRuntime->reloLogger(), 5, "\tpreparePrivateData: ramMethod %p failValidation %d\n", ramMethod, failValidation);
+   reloPrivateData->_failValidation = !inlinedSiteIsValid;
+   RELO_LOG(reloRuntime->reloLogger(), 5, "\tpreparePrivateData: ramMethod %p inlinedSiteIsValid %d\n", ramMethod, inlinedSiteIsValid);
    }
 
 
 bool
-TR_RelocationRecordInlinedMethod::validateClassesSame(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget, TR_OpaqueMethodBlock **theMethod)
+TR_RelocationRecordInlinedMethod::inlinedSiteValid(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget, TR_OpaqueMethodBlock **theMethod)
```

Comment:
Both this method and its caller (`preparePrivateData`) are working with `J9Method*`. Perhaps this should accept `J9Method **theMethod` instead of `TR_OpaqueMethodBlock**`?

Reply:
It turns out that `preparePrivateData` initialized a `J9Method` but always worked with a `TR_OpaqueMethodBlock`, so I just updated `preparePrivateData` to use an opaque method block.